code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/OptimizeExtension.java

```diff
@@ -628,34 +656,94 @@ void handleEdgeResponse(@NonNull final Event event) {
      * @param event incoming {@link Event} object to be processed.
      */
     void handleEdgeErrorResponse(@NonNull final Event event) {
-        if (OptimizeUtils.isNullOrEmpty(event.getEventData())) {
-            Log.debug(
+        try {
+            final Map<String, Object> eventData = event.getEventData();
+            final String requestEventId = OptimizeUtils.getRequestEventId(event);
+
+            if (!OptimizeUtils.isEdgeErrorResponseContent(event)
```

Comment:
nit: Why do we need to verify if this event `isEdgeErrorResponseContent`? The listener is only called for event type Edge and source response content. This check is superfluous